### ISSUE TITLE
GHC 8.10.7: Failed to get project GHC version:CradleError

### ISSUE THREAD

**Issue by bingis-khan** [haskell/haskell-language-server#2495]
### Your environment
Windows 10
Visual Studio Code
Stack Project - one created with `stack new [...] simple`

Everything installed with ghcup.

### Steps to reproduce
Set GHC version to GHC 8.10.7
Create a `stack new [name] simple` project.
Watch the hls return with an error five times.

### Expected behaviour
It should not crash / properly boot. This happens when I downgrade to 8.8.4 (might work with others, I haven't tested it that much yet).

### Actual behaviour
HLS fails to run with an error: Failed to get project GHC version:CradleError {cradleErrorDependencies = [], cradleErrorExitCode = ExitFailure 1, cradleErrorStderr = ["Error when calling stack setup --silent","",""]}

### Include debug information
Execute in the root of your project the command `haskell-language-server-wrapper --debug .` and paste the logs here:
(if you are using the vscode extension check the executable location [here](https://github.com/haskell/vscode-haskell#downloaded-binaries))
<details>
<summary>
Debug output:
</summary>

```
PS C:\Users\bingis_khan\Desktop\now_doin\haskell\haskell_other_src\BobFunc> haskell-language-server-wrapper --debug
No 'hie.yaml' found. Try to discover the project type!
Run entered for haskell-language-server-wrapper(haskell-language-server-wrapper-1.5.1.exe) Version 1.5.1.0, Git revision 745ef26f406dbdd5e4a538585f8519af9f1ccb09 (dirty) x86_64 ghc-8.10.7
Current directory: C:\Users\bingis_khan\Desktop\now_doin\haskell\haskell_other_src\BobFunc
Operating system: mingw32
Arguments: ["--debug"]
Cradle directory: C:\Users\bingis_khan\Desktop\now_doin\haskell\haskell_other_src\BobFunc
Cradle type: Stack

Tool versions found on the $PATH
cabal:          3.6.2.0
stack:          2.7.3
ghc:            8.10.7


Consulting the cradle to get project GHC version...
Failed to get project GHC version:CradleError {cradleErrorDependencies = [], cradleErrorExitCode = ExitFailure 1, cradleErrorStderr = ["Error when calling stack setup --silent","",""]}
```
</details>

tl; dr
Seems to only happen with GHC 8.10.7 - downgrading to 8.8.4 (LTS 16.27) helps.
Non-Stack source files are checked correctly.
New Windows & Haskell (GHC) installation.
Haven't found a similar issue or a solution.

**Comment by fendor** [comment]
Hello, thank you for the bug report!

Can you tell us what the output of `stack setup --silent` is?

**Comment by bingis-khan** [comment]
@fendor 
Sure thing.
`stack setup --silent` is (rather obviously):
```
PS C:\Users\bingis_khan\Desktop\now_doin\haskell\haskell_other_src\BobFunc> stack setup --silent
PS C:\Users\bingis_khan\Desktop\now_doin\haskell\haskell_other_src\BobFunc>   
```
... silent.

But `stack setup` yields this:
```
PS C:\Users\bingis_khan\Desktop\now_doin\haskell\haskell_other_src\BobFunc> stack setup
The GHC located at C:\Users\bingis_khan\AppData\Local\Programs\stack\x86_64-windows\ghc-8.10.7\bin\ghc-8.10.7.exe failed to compile a sanity check. Please see:

    http://docs.haskellstack.org/en/stable/install_and_upgrade/

for more information. Exception was:
Received ExitFailure 1 when running
Raw command: "C:\\Users\\bingis_khan\\AppData\\Local\\Programs\\stack\\x86_64-windows\\ghc-8.10.7\\bin\\ghc-8.10.7.exe" "C:\\Users\\bingis_khan\\AppData\\Local\\Temp\\stack-sanity-check-b99f8b5eb2c490ee\\Main.hs" -no-user-package-db
Run from: C:\Users\bingis_khan\AppData\Local\Temp\stack-sanity-check-b99f8b5eb2c490ee\
Standard output:

Loaded package environment from C:\Users\bingis_khan\AppData\Roaming\ghc\x86_64-mingw32-8.10.7\environments\default
[1 of 1] Compiling Main             ( C:\\Users\bingis_khan\AppData\Local\Temp\stack-sanity-check-b99f8b5eb2c490ee\Main.hs, C:\\Users\bingis_khan\AppData\Local\Temp\stack-sanity-check-b99f8b5eb2c490ee\Main.o )
Standard error:


C:\\Users\bingis_khan\AppData\Local\Temp\stack-sanity-check-b99f8b5eb2c490ee\Main.hs:1:1: error:
    Could not load module `Distribution.Simple'
    It is a member of the hidden package `Cabal-3.2.1.0'.
    You can run `:set -package Cabal' to expose it.
    (Note: this unloads all the modules in the current scope.)
    Use -v (or `:set -v` in ghci) to see a list of the files searched for.
  |
1 | import Distribution.Simple
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^
```

**Comment by fendor** [comment]
Did you by any chance ever execute `cabal install --lib`? Looks like it to me. 

Anyway, look at the contents of `C:\Users\bingis_khan\AppData\Roaming\ghc\x86_64-mingw32-8.10.7\environments\default`, I think it contains some package hashes.

Easiest fix is to remove the file `C:\Users\bingis_khan\AppData\Roaming\ghc\x86_64-mingw32-8.10.7\environments\default`. After that, `stack setup` should succeed, imo.

**Comment by bingis-khan** [comment]
@fendor Yeah, that did it!
But, is there a replacement for doing `cabal install --lib`?
I generally use stack for bigger projects and "naked" GHC/Cabal for small, one-off programs. Both are equally important. Being able to access Haskell packages from anywhere is just too convenient, unlike a "global project".

**Comment by jneira** [comment]
> @fendor Yeah, that did it! 

nice!

> But, is there a replacement for doing `cabal install --lib`? I generally use stack for bigger projects and
 "naked" GHC/Cabal for small, one-off programs. Both are equally important. Being able to access Haskell packages from anywhere is just too convenient, unlike a "global project".

there are several ways to avoid `cabal install --lib` and run haskell code needing external packages in scope:

- `cabal repl -b package`: to get a repl with a concrete package or packages in scope
- `cabal/stack run standalonescript.hs`: to run a script which can use packages 

there are ways to make `cabal install --lib` less harmful

In any case i think we can close the issue, feel free to reopen, thanks for the report!

**Comment by kintrix007** [comment]
Hey, I have the same issue. I have also (at some point) ran `cabal install --lib` not realizing it is harmful. I am, however, using Linux (Fedora Workstation).

@jneira, is there any chance you could describe how you found the file to delete? Or even better, could you tell me where that file could be under linux?

**Comment by bingis-khan** [comment]
@kintrix007 iirc, it should be under .cabal in your $HOME

**Comment by kintrix007** [comment]
> @kintrix007 iirc, it should be under .cabal in your $HOME

Trying to find files/directories named default yields only two results there, neither of which looks like a default environment..?

```sh
$ find $HOME/.cabal -name "default"
/home/kintrix/.cabal/store/ghc-8.10.7/fourmolu-0.3.0.0-4dd5e2e762a45933075bc92fbb6d2c7b8cfd39df26b79e7cc725e0722f5bc333/share/data/examples/declaration/default
/home/kintrix/.cabal/store/ghc-8.10.7/ormolu-0.1.4.1-a4aa4b1f7d3e82b3ce68dae9c91a57983f2adeb46997962da591ec6cec46b7d8/share/data/examples/declaration/default
```
Thanks for the quick reply, though!
